src/compiler/checker/lines_18000_20000.rs

```diff
@@ -321,8 +321,17 @@ impl<'type_checker> CheckTypeRelatedTo<'type_checker> {
         &self,
         source: Rc<Type>,
         target: &UnionOrIntersectionType,
+        report_errors: bool,
     ) -> Ternary {
         let target_types = target.types();
+        if target.flags().intersects(TypeFlags::Union) {
+            if self
+                .type_checker
+                .contains_type(target_types, source.clone())
```

Comment:
I guess the simple case for a "yes is assignable" to a union-type target is when the source "is" one of the union'ed target types (based on a type ID comparison), which is the case here since string literal types are cached to be unique